src/Emulator/Instructions/RestartModal.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright (C) 1992-2021 Free Software Foundation, Inc.
+
+This file is part of ToyNet React.
+
+ToyNet React is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License as published by the Free
+Software Foundation; either version 3, or (at your option) any later
+version.
+
+ToyNet React is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with ToyNet React; see the file LICENSE.  If not see
+<http://www.gnu.org/licenses/>.
+
+*/
+import React from 'react';
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalBody,
+  ModalFooter,
+  Button,
+  Heading,
+  Text,
+} from '@chakra-ui/core';
+import { useDialogue } from '../../common/providers/EmulatorProvider';
+
+interface Props {
+  isOpen: boolean;
+
+  close: () => any;
+}
+
+const RestartModal = ({ isOpen, close }: Props) => {
+  const { clearDialogue } = useDialogue();
+
+  /*
+  Triggered by onClick of I'm sure button on the modal.
+  It will clear action logs and updates of emulators from session storage. Then reload the page.
+  */
+  const handleRestart = () => {
+      clearDialogue();
```

Comment:
Why do we need to clear the dialogue here? This should be handled when session storage is clear, I think.

Reply:
I forgot to remove this line which I added when I was playing to know how EmulatorProvider is working in this app!

Reply:
Perfect!